src/StackExchange.Redis/ConfigurationOptions.cs

```diff
@@ -792,21 +792,21 @@ private void DoParse(string configuration, bool ignoreUnknown)
 
         internal bool IsAzureEndpoint()
         {
+            List<string> azureRedisHosts = new List<string> {
+                ".redis.cache.windows.net",
+                ".redis.cache.chinacloudapi.cn",
+                ".redis.cache.usgovcloudapi.net",
+                ".redis.cache.cloudapi.de",
+                ".redisenterprise.cache.azure.net"
+            };
+
             foreach (var ep in EndPoints)
```

Comment:
Can we replace this `foreach` with `Linq` too?

Reply:
I'm always a little cautious of blindly applying LINQ; you tend to end up with unexpected closures; see `dnsEp` in `Any`, for example. I'm not saying "never use LINQ", but: if the existing code works fine and is clear: I wouldn't reach for LINQ, either